libs/dokobit-signing/src/lib/signing.service.ts

```diff
@@ -0,0 +1,159 @@
+import { RESTDataSource, RequestOptions } from 'apollo-datasource-rest'
+import { Base64 } from 'js-base64'
+import { createHash } from 'crypto'
+
+import { Injectable, Inject } from '@nestjs/common'
+
+import { Logger, LOGGER_PROVIDER } from '@island.is/logging'
+
+export interface SigningServiceOptions {
+  url: string
+  accessToken: string
+}
+
+export class SigningServiceResponse {
+  controlCode: string
+  documentToken: string
+}
+
+export class SigningServiceError extends Error {
+  constructor(code: number, message: string) {
+    super()
+
+    this.name = 'DokobitError'
+    this.message = `${code}: ${message}`
+  }
+}
+
+interface DokobitSignResponse {
+  status: string
+  control_code: string
+  token: string
+  error_code: number
+  message: string
+}
+
+interface DokobitStatusResponse {
+  status: string
+  file: {
+    name: string
+    digest: string
+    content: string
+  }
+  signature_id: string
+  error_code: number
+  message: string
+}
+
+@Injectable()
+export class SigningService extends RESTDataSource {
+  constructor(
+    @Inject('SIGNING_OPTIONS') private options: SigningServiceOptions,
+    @Inject(LOGGER_PROVIDER)
+    private logger: Logger,
+  ) {
+    super()
+
+    this.baseURL = `${this.options.url}/mobile/`
+  }
+
+  willSendRequest(request: RequestOptions) {
+    request.headers.set('Content-Type', 'multipart/form-data')
+  }
+
+  async requestSignature(
+    mobileNumber: string,
+    message: string,
+    contact: string,
+    location: string,
+    documentName: string,
+    documentContent: string,
+  ): Promise<SigningServiceResponse> {
+    this.logger.debug(
+      `${contact} ${location} is requesting ${mobileNumber} to sign ${documentName} with message ${message}`,
+    )
+
+    const base64 = Base64.encode(documentContent)
+    const digest = createHash('sha256')
+      .update(base64)
+      .digest('hex')
+
+    const body = new FormData()
+    body.append('phone', mobileNumber)
+    body.append('message', `${message} `)
```

Comment:
Please consider replacing 

```
 body.append('message', `${message} `)
```

with
 
```
 body.append('message', message)
```

for code readability

Reply:
I need the extra space at the end. A four digit security code is added to the end of this message on the mobile phone. Without the space, the display will be kind of ugly.